arangod/Agency/RestAgencyHandler.cpp

```diff
@@ -181,13 +181,22 @@ RestStatus RestAgencyHandler::pollIndex(
             builder.add("code", VPackValue(int(ResponseCode::OK)));
             builder.add(VPackValue("result"));
             VPackObjectBuilder r(&builder);
-            auto const firstIndex = slice.get("firstIndex");
+            if (!slice.get("firstIndex").isNumber()) {
+              generateError(
```

Comment:
Does this branch not need a `return` statement? Otherwise we fall through and continue...